#goal definition
char[] inputContent
char[] inputMode
---
#result definition
char[] returnMessage
---
#feedback
int32[] sequence
